src/mach.js

```diff
@@ -100,6 +239,13 @@ const mach = {
     return cv.canvas.height;
   },
 
+  machEventShift() {
+    if (mach.events.length < 0)
+      return 0;
+
+    return mach.events.shift();
```

Comment:
Note to self: Possible optimization: Use an array and a cursor indicating the current head of the list. Each time we call this function, return list[cursor] and then increment it. Finally reset the list and cursor at the end of frame (needs a new function)

The point is that shift operation may involve moving of all array elements which is performance intensive. In practice, it is likely that JS engines optimize this properly.